From 9642aea386dafc43c0e68964c3310d6cd340fe3b Mon Sep 17 00:00:00 2001
Message-ID: <9642aea386dafc43c0e68964c3310d6cd340fe3b.1710967275.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1710967273.git.sharlatanus@gmail.com>
References: <cover.1710967273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 12 Mar 2024 22:26:57 +0000
Subject: [PATCH 45/49] gnu: Add python-ewah-bool-utils.

* gnu/packages/python-compression.scm (python-ewah-bool-utils): New variable.

Change-Id: Ib9730d599f66bc2b864bfd305135e367f89ba9dd
---
 gnu/packages/python-compression.scm | 35 +++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/gnu/packages/python-compression.scm b/gnu/packages/python-compression.scm
index d076a4e161..16e9fc7a4b 100644
--- a/gnu/packages/python-compression.scm
+++ b/gnu/packages/python-compression.scm
@@ -196,6 +196,41 @@ (define-public python-cramjam
 @end itemize")
     (license license:expat)))
 
+(define-public python-ewah-bool-utils
+  (package
+    (name "python-ewah-bool-utils")
+    (version "1.1.0")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "ewah_bool_utils" version))
+       (sha256
+        (base32 "11z9vv9m19370j762z100dqhy2ndjzkxk3w9z02lzwadbyibal3h"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'build-extensions
+            (lambda _
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+    (native-inputs
+     (list python-cython python-pytest))
+    (propagated-inputs
+     (list python-numpy))
+    (home-page "https://github.com/yt-project/ewah_bool_utils")
+    (synopsis "EWAH Bool Array compression")
+    (description "This package provide a Python wrapper to @acronym{EWAH,
+ Enhanced Word-Aligned Hybrid} compression bitarray method.")
+    (license (list
+              ;; LICENSE: for Python code.
+              license:bsd-3
+              ;; XXX: Check if it's possible to have it as a dedicated package
+              ;; and link here..
+              ;; ewah_bool_utils/cpp/LICENSE: for C++ bundle, sourced from
+              ;; <https://github.com/lemire/EWAHBoolArray>.
+              license:asl2.0))))
+
 (define-public python-pybcj
   (package
     (name "python-pybcj")
-- 
2.41.0

